diff -Naur a/runtime/src/qdrant.rs b/runtime/src/qdrant.rs
--- a/runtime/src/qdrant.rs	2026-04-17 19:30:13.000000000 +0200
+++ b/runtime/src/qdrant.rs	2026-04-23 10:46:49.724719408 +0200
@@ -134,7 +134,7 @@
     ]);
     
     let server_spawn_clone = QDRANT_SERVER.clone();
-    let qdrant_relative_source_path = "resources/databases/qdrant/config.yaml";
+    let qdrant_relative_source_path = "/app/databases/qdrant/config.yaml";
     let qdrant_source_path = match path_resolver.resolve_resource(qdrant_relative_source_path) {
         Some(path) => path,
         None => {
@@ -306,4 +306,4 @@
         }
     }
     Ok(())
-}
\ No newline at end of file
+}
